packages/collection/src/collection.ts

```diff
@@ -662,6 +662,21 @@ export class Collection<K, V> extends Map<K, V> {
 		return coll;
 	}
 
+	/**
+	 * The complement method returns a new structure containing items where the keys and values of other structure are not present in the original.
+	 *
+	 * @param other - The other Collection to filter against
+	 */
+	public complement<T>(other: ReadonlyCollection<K, T>): Collection<K, T> {
+		const coll = new this.constructor[Symbol.species]<K, T>();
+		for (const [k, v] of other) {
+			if (!this.has(k) && !Object.is(v, this.get(k))) {
+				coll.set(k, v);
+			}
+		}
```

Comment:
What should `original.complement(other)` return?
<details>
  <summary>1</summary>

![image](https://user-images.githubusercontent.com/62868468/181825384-2575063f-47ea-426c-919c-6a6b3fae3efd.png)

</details>
<details>
  <summary>2 (currently)</summary>

![image](https://user-images.githubusercontent.com/62868468/181825303-8b58937c-02eb-4833-bdf4-f1c7f950e269.png)

</details>

Reply:
I'd prefer it to be `a.minus(b)` and result 1 myself. Simpler that way.

Reply:
This was minus also sounds nicer and easier to grasp 👀